feat: statistics calculator (#48181)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-advanced-array-methods-by-building-a-statistics-calculator/635060a5c03c950f46174cb5.md

---
id: 635060a5c03c950f46174cb5
title: Step 1
challengeType: 0
dashedName: step-1
---

# --description--

> Statistics is an important subset of mathematics that focuses on analyzing data for patterns and trends, and using that information to make predictions and draw conclusions.

Reviewer: ```suggestion
Statistics is a way of using math to make sense of data. It helps us understand patterns and trends in information, so we can make predictions and make decisions based on that information.
```

In this challenge, you will build a statistics calculator that takes a set of numbers and returns the mean, median, mode, standard deviation, and variance.

> The HTML and CSS has been provided for you. Feel free to explore the code - you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.

Reviewer: ```suggestion
The HTML and CSS have been provided for you. Feel free to explore the code - you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.
```

> The HTML and CSS have been provided for you. Feel free to explore the code - you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.

Reviewer 2: Hyphen to en dash:

```suggestion
The HTML and CSS have been provided for you. Feel free to explore the code – you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.
```

> The HTML and CSS have been provided for you. Feel free to explore the code - you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.

Reviewer 2: Hyphen to en dash:

```suggestion
The HTML and CSS have been provided for you. Feel free to explore the code – you may notice that the `calculate` function is called when the form is submitted. When you are ready, declare a `calculate` variable and assign it an empty function in the `script.js` file.
```
